drivers/serial/uart_nrfx_uarte.c

```diff
@@ -0,0 +1,644 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Driver for Nordic Semiconductor nRF52 UARTE
+ */
+
+
+#include <uart.h>
+#include <gpio.h>
+#include <hal/nrf_uarte.h>
+
+/* Device data structure */
+typedef struct {
+	u8_t rx_data;
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_callback_t cb;	/**< Callback function pointer */
+	u8_t tx_buffer[CONFIG_UART_NRFX_RAM_BUFFER_SIZE];
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+} uarte_nrfx_data_t;
+
+typedef struct
+{
+	NRF_UARTE_Type * p_reg;        /* Pointer to registers structure */
+	uint8_t          drv_inst_idx; /* Driver instance index */
```

Comment:
What is the purpose of this? A value is assigned to this field but it is never referenced.